app/src/main/java/com/eveningoutpost/dexdrip/utils/LibreTrendGraph.java

```diff
@@ -94,30 +60,24 @@ public void closeNow(View view) {
         }
         
         LibreTrendLatest libreTrendLatest = LibreTrendUtil.getInstance().getLibreTrendLatest();
-        if(libreTrendLatest.glucoseLevelRaw == 0) {
-            Log.e(TAG, "libreBlock exists but libreTrendLatest.glucoseLevelRaw is zero ");
+        if(libreTrendLatest == null) {
+            Log.e(TAG, "LibreTrendPoints exists but LibreTrendUtil is NULL.");
             return null;
         }
         ArrayList<Float> ret = new ArrayList<Float>();
         
-        double factor = libreTrendLatest.bg / libreTrendLatest.glucoseLevelRaw;
+        double factor = libreTrendLatest.getFactor();
         if(factor == 0) {
-            // We don't have the calculated value, but we do have the raw value. (No calibration exists)
-            // I want to show raw data.
-            Log.w(TAG, "Bg data was not calculated, working on raw data");
-            List<BgReading> latestReading = BgReading.latestForGraph (1, libreTrendLatest.timestamp - 1000, libreTrendLatest.timestamp + 1000);
-            if(latestReading == null || latestReading.size() == 0) {
-                Log.e(TAG, "libreBlock exists but no matching bg record exists");
-                return null;
-            }
-            
-            factor = latestReading.get(0).raw_data / libreTrendLatest.glucoseLevelRaw;
+            Log.e(TAG, "getLatestBgForXMinutes factor is null returning.");
+            return null;
+
         }
         
         int count = 0;
         for(int i = libreTrendLatest.id ; i >= 0 && count < NumberOfMinutes; i--) {
             count ++;
             ret.add(new Float(factor * LibreTrendPoints.get(i).rawSensorValue));
+            //Log.e(TAG, "getLatestBgForXMinutes adding a point to the graph.");
```

Comment:
Please do not comment out code if it is not necessary anymore. Remove it, instead!

Reply:
Done.